### PR TITLE
fix toShortSafely: yes we can safely convert -1 to short :)

### PR DESCRIPTION
**Pull request by mpollmeier**
also: fix array kind bounds check

### CODE REVIEW

File: core/src/main/scala/flatgraph/Graph.scala [hunk 1]
```diff
@@ -52,7 +52,7 @@ class Graph(val schema: Schema, val storagePathMaybe: Option[Path] = None) exten
 
   /** Note: this included `deleted` nodes! You might want to use `livingNodeCountByKind` instead. */
   private[flatgraph] def nodeCountByKind(kind: Int): Int =
-    if (nodesArray.length <= kind) 0
+    if (nodesArray.length < kind || kind < 0) 0
```
Comment (bbrehm) [inline core/src/main/scala/flatgraph/Graph.scala:55]: The `kind == nodesArray.length` case is off-by-one and should be included in this check -- this ain't julia ;)

I would remove this check entirely: it is clearly an error and an immediate array-oob exception is best for debugging. But if we have the check then we should do it right.

File: core/src/main/scala/flatgraph/misc/Conversions.scala [hunk 2]
```diff
@@ -3,7 +3,10 @@ package flatgraph.misc
 object Conversions {
   extension (i: Int) {
     def toShortSafely: Short = {
-      assert(0 <= i && i <= Short.MaxValue, throw new ConversionException(s"cannot safely downcast int with value=$i to short"))
+      assert(
```
Comment (bbrehm) [inline core/src/main/scala/flatgraph/misc/Conversions.scala:6]: Again, I would not include such checks. But if you want to include it, then I guess `i == i.toShort.toInt` is more to the point.

**Review by bbrehm — APPROVED** [review]

File: core/src/main/scala/flatgraph/Graph.scala (hunk 1 above)
Comment (mpollmeier) [reply core/src/main/scala/flatgraph/Graph.scala:55]: whoopsie, thanks for catching
the reason for introducing this check is to provide a more specific error than an array-oob exception which is quite generic

Comment (mpollmeier) [reply core/src/main/scala/flatgraph/Graph.scala]: Updated - I also change the semantics: we now throw an exception if someone asks for an invalid kind, previously that would just return 0. I'm still undecided which way to go tbh... do you have a preference?

File: core/src/main/scala/flatgraph/misc/Conversions.scala (hunk 2 above)
Comment (mpollmeier) [reply core/src/main/scala/flatgraph/misc/Conversions.scala]: The reason for introducing this check is to provide a more specific error than an array-oob exception which is quite generic. 
The `.toShort.toInt` check is equivalent but more clear, so yes good suggestion, thanks

Comment (bbrehm) [reply core/src/main/scala/flatgraph/misc/Conversions.scala]: >The reason for introducing this check is to provide a more specific error than an array-oob exception which is quite generic.

Yeah, but this is not an error condition that anyone is supposed to specifically catch and recover from. 

It is an exception where somebody is supposed to stare at the stack-trace in an error log, look at the line numbers, and also look at the relevant code path. 

Given that the person handling the exception already has that info, most manual throw error strings are pretty superfluous: For such exceptions, a source-code comment around the `throw` is equivalent to the static string part of an error message.

**Pull request merged**